### PR TITLE
#714 FIX

### PR DESCRIPTION
**Pull request by TikhomirovSergey**
## Change list

- Update to Selenium 3.5.3
- phantomJSdriver and htmlunit-drive were excluded
- eclipse compiler was updated to 4.6.1
- also AppiumElementLocator was improved

 
## Types of changes

- [ ] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

Dependencies of the html unit driver
https://mvnrepository.com/artifact/org.seleniumhq.selenium/htmlunit-driver/2.27

Dependencies of the phantom js driver
https://mvnrepository.com/artifact/com.codeborne/phantomjsdriver/1.4.3

Also something weird was foaund in AppiumElementLocator. I wonder why we couldn't find it before.

### CODE REVIEW

File: src/main/java/io/appium/java_client/pagefactory/AppiumElementLocator.java [hunk 1]
```diff
@@ -21,8 +21,6 @@
 import static io.appium.java_client.pagefactory.ThrowableUtil.isStaleElementReferenceException;
 
 
-import com.google.common.base.Function;
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/pagefactory/AppiumElementLocator.java]: I also see this import in several other sources - are they all expected?

File: build.gradle [hunk 2]
```diff
@@ -54,14 +54,18 @@ compileJava {
     ]
 }
 
-ext.seleniumVersion = '[3.5.2,3.5.2]'
+ext.seleniumVersion = '[3.5.3,3.5.3]'
```
Comment (mykola-mokhnach) [inline build.gradle]: what is the point to set a range, which includes only one item? Isn't it simply equal to '3.5.3'?

**Comment by mykola-mokhnach** [comment]
I was also thinking about enforcing the version of guava library, since our module cannot work if it is older than 20

File: build.gradle (hunk 2 above)
Comment (TikhomirovSergey) [reply build.gradle]: @mykola-mokhnach The purpose is to make a valid pom.xml file. The version like 3.5+ is malformed for maven.

**Comment by SrinivasanTarget** [comment]
@TikhomirovSergey As @mykola-mokhnach suggested i think we should force `com.google.guava:guava:23.0` dependency.

File: src/main/java/io/appium/java_client/pagefactory/AppiumElementLocator.java (hunk 1 above)
Comment (TikhomirovSergey) [reply src/main/java/io/appium/java_client/pagefactory/AppiumElementLocator.java]: @mykola-mokhnach No
Except the `AppiumFunction` class.

**Comment by TikhomirovSergey** [comment]
@SrinivasanTarget I will try it with the `com.google.guava:guava:23.0` dependency.

File: src/test/java/io/appium/java_client/android/OpenNotificationsTest.java [hunk 3]
```diff
@@ -17,18 +16,15 @@ public void openNotification() throws Exception {
...
+        assertNotEquals(0, wait.until(input -> {
+            List<AndroidElement> result = input
+                    .findElements(id("com.android.systemui:id/carrier_label"));
 
-                if (result.size() == 0) {
-                    return null;
-                }
-
-                return result;
+            if (result.size() == 0) {
```
Comment (mykola-mokhnach) [inline src/test/java/io/appium/java_client/android/OpenNotificationsTest.java]: return result.isEmpty() ? null : result;

**Comment by TikhomirovSergey** [comment]
@SrinivasanTarget I think it is not necessary to force the `com.google.guava:guava:23.0` because Selenium already depends on this version.

https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-remote-driver/3.5.3

**Comment by SrinivasanTarget** [comment]
> I think it is not necessary to force the com.google.guava:guava:23.0 because Selenium already depends on this version.

Ok

**Pull request merged**